pallets/parachain-staking/src/rewards/transfer_from_rewards_account.rs

```diff
@@ -0,0 +1,212 @@
+use crate::{
+	traits::PayoutReward, BalanceOf, Config, Error, Event, Pallet, RewardsAccount, RoundIndex,
+};
+use frame_support::{
+	ensure,
+	pallet_prelude::Weight,
+	traits::tokens::{currency::Currency, ExistenceRequirement},
+};
+use sp_runtime::{traits::Zero, ArithmeticError, DispatchError};
+
+pub struct TransferFromRewardsAccount;
+impl<Runtime: crate::Config> PayoutReward<Runtime> for TransferFromRewardsAccount {
+	fn payout_collator_rewards(
+		for_round: crate::RoundIndex,
+		collator_id: Runtime::AccountId,
+		amount: crate::BalanceOf<Runtime>,
+	) -> Weight {
+		crate::Pallet::<Runtime>::send_collator_reward(for_round, collator_id, amount)
+	}
+
+	fn payout(
+		delegator_id: &Runtime::AccountId,
+		amount: crate::BalanceOf<Runtime>,
+	) -> Result<crate::BalanceOf<Runtime>, DispatchError> {
+		// Early return if amount is zero,
+		if amount.is_zero() {
+			return Ok(Zero::zero());
+		}
+
+		// Early return if RewardsAccount is not set.
+		let rewards_account = match RewardsAccount::<Runtime>::get() {
+			Some(account) => account,
+			None => return Ok(Zero::zero()),
+		};
+
+		// Ensure the destination account exists with a clearer error message.
+		ensure!(
+			frame_system::Account::<Runtime>::contains_key(delegator_id),
+			Error::<Runtime>::DeadAccount
+		);
+
+		// Directly handle the result of the transfer, making use of match
+		// for clearer error handling.
+		match Runtime::Currency::transfer(
+			&rewards_account,
+			&delegator_id,
+			amount,
+			ExistenceRequirement::KeepAlive,
+		) {
+			Ok(_) => Ok(amount),
+			Err(DispatchError::Arithmetic(ArithmeticError::Underflow)) => Ok(Zero::zero()),
+			Err(e) => Err(e),
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn send_collator_reward(
+		_round_idx: RoundIndex,
+		collator_id: T::AccountId,
+		amt: BalanceOf<T>,
```

Comment:
```suggestion
		amount: BalanceOf<T>,
```
to keep the naming consistent